mrp_cut/models/mrp_bom_line.py

```diff
@@ -1,55 +1,74 @@
 # Copyright 2021 Berezi Amubieta - AvanzOSC
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
-from odoo import models, fields, api, _
+from odoo import _, api, fields, models
 from odoo.exceptions import ValidationError
 
 
 class MrpBomLine(models.Model):
     _inherit = "mrp.bom.line"
 
-    long_cut = fields.Float(string='Long Cut (mm)', default=0)
+    long_cut = fields.Float(
+        string="Long Cut (mm)",
+        default=0,
+    )
     qty_pieces_set = fields.Integer(
-        string='Quantity of Pieces in Set', default=1)
+        string="Quantity of Pieces in Set",
+        default=1,
+    )
     waste_rate = fields.Float(
-        string='Waste Rate', default=1.0)
+        string="Waste Rate",
+        default=1.0,
+    )
     qty_second_uom = fields.Float(
-        string='Quantity in Second UOM', compute='_compute_qty_second_uom',
-        store=True)
+        string="Quantity in Second UOM",
+        compute="_compute_qty_second_uom",
+        store=True,
+    )
     second_uom_id = fields.Many2one(
-        string='Second UOM', comodel_name='uom.uom',
-        related='product_id.second_uom_id', store=True)
+        string="Second UOM",
+        comodel_name="uom.uom",
+        related="product_id.second_uom_id",
+        store=True,
+    )
     pieces_second_uom = fields.Float(
-        string='Pieces second UOM', compute='_compute_pieces_second_uom',
-        store=True)
+        string="Pieces second UOM",
+        compute="_compute_pieces_second_uom",
+        store=True,
+    )
     read_only = fields.Boolean(
-        string='Read Only', compute='_compute_read_only')
+        string="Read Only",
+        compute="_compute_read_only",
+    )
 
-    @api.onchange('long_cut')
+    @api.onchange("long_cut")
     def onchange_waste_rate(self):
         if self.long_cut:
             if self.long_cut <= 1000:
                 self.waste_rate = 1.08
-            if self.long_cut > 1000 and self.long_cut <= 2300:
+            elif self.long_cut > 1000 and self.long_cut <= 2300:
                 self.waste_rate = 1.12
-            if self.long_cut > 2300 and self.long_cut <= 3000:
+            elif self.long_cut > 2300 and self.long_cut <= 3000:
                 self.waste_rate = 1.15
-            if self.long_cut > 3000:
+            elif self.long_cut > 3000:
                 self.waste_rate = 1.2
+            else:
```

Comment:
No es correcto. Al else no va a entrar nunca, se supone, pero si entrase el valor no sería el de menos de 1000, sino el de más de 3000.

Reply:
if self.long_cut <= 1000:
                self.waste_rate = 1.08
            elif self.long_cut > 1000 and self.long_cut <= 2300:
                self.waste_rate = 1.12
            elif self.long_cut > 2300 and self.long_cut <= 3000:
                self.waste_rate = 1.15
            elif self.long_cut > 3000:
                self.waste_rate = 1.2
            else:
                self.waste_rate = 1.0

no puede ser menor que cero. Siempre estará entre 0 e infinito. Si es mayor de 3000, self.waste_rate sería 1.2, no puede ser 1.0. 
Para mí es más correcto poner
  if self.long_cut <= 1000:
                self.waste_rate = 1.08
            elif self.long_cut > 1000 and self.long_cut <= 2300:
                self.waste_rate = 1.12
            elif self.long_cut > 2300 and self.long_cut <= 3000:
                self.waste_rate = 1.15
            else:
                self.waste_rate = 1.2

Que sería el caso > 3000.